packages/editor/src/toolbar/tools/colors.tsx

```diff
@@ -82,6 +100,16 @@ export function ColorTool(props: ColorToolProps) {
               onColorChange();
               config.set(cacheKey, null);
             }}
+            onSave={() => {
+              if (!activeColor) return;
+              const colors = config.get<string[]>(`custom_${cacheKey}`, []);
+              if (colors?.includes(config.get(cacheKey)!)) return;
+              if (DEFAULT_COLORS.includes(config.get(cacheKey)!)) return;
+
+              colors?.push(config.get(cacheKey)!);
```

Comment:
Instead of getting the current color from `config` why not take it as a parameter in the `onSave` function?